src/liballoc/collections/btree/node.rs

```diff
@@ -463,13 +463,13 @@ impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::Immut<'a>, K, V, Type> {
     /// SAFETY: the caller must ensure that the node has more than `idx` elements.
     unsafe fn into_key(self, idx: usize) -> &'a K {
         debug_assert!(idx < self.len());
-        unsafe { &*MaybeUninit::first_ptr(&(*self.as_leaf()).keys).add(idx) }
+        unsafe { &MaybeUninit::get_ref(&(*self.as_leaf()).keys[idx]) }
     }
 
     /// SAFETY: the caller must ensure that the node has more than `idx` elements.
     unsafe fn into_val(self, idx: usize) -> &'a V {
         debug_assert!(idx < self.len());
-        unsafe { &*MaybeUninit::first_ptr(&(*self.as_leaf()).vals).add(idx) }
+        unsafe { &MaybeUninit::get_ref(&(*self.as_leaf()).vals[idx]) }
```

Comment:
Nice!
Could you add a comment saying that we are only creating a reference to the one element we are interested in here, and thus avoid aliasing with outstanding references to other elements?